From 788316932dbfaa653a3c1c332ded2a996d20c475 Mon Sep 17 00:00:00 2001
From: Heiko Stuebner <heiko@sntech.de>
Date: Fri, 8 Dec 2017 13:29:42 +0100
Subject: [PATCH] drm/lima: Set gpu_type from devicetree match data

Later Mali Utgard like the Mali450 require some slightly different handling
so it is required to set the gpu-type accordingly. We already know the
type from the compatible, so we should just set the type from the attached
data.

Signed-off-by: Heiko Stuebner <heiko@sntech.de>
---
 drivers/gpu/drm/lima/lima_device.c | 2 --
 drivers/gpu/drm/lima/lima_drv.c    | 3 ++-
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/lima/lima_device.c b/drivers/gpu/drm/lima/lima_device.c
index eb6902f..05f25f7 100644
--- a/drivers/gpu/drm/lima/lima_device.c
+++ b/drivers/gpu/drm/lima/lima_device.c
@@ -174,8 +174,6 @@ int lima_device_init(struct lima_device *ldev)
 
 	dma_set_coherent_mask(ldev->dev, DMA_BIT_MASK(32));
 
-	ldev->gpu_type = GPU_MALI400;
-
 	np = ldev->dev->of_node;
 
 	err = lima_clk_init(ldev);
diff --git a/drivers/gpu/drm/lima/lima_drv.c b/drivers/gpu/drm/lima/lima_drv.c
index 9896b80..50fd078 100644
--- a/drivers/gpu/drm/lima/lima_drv.c
+++ b/drivers/gpu/drm/lima/lima_drv.c
@@ -235,6 +235,7 @@ static int lima_pdev_probe(struct platform_device *pdev)
 
 	ldev->pdev = pdev;
 	ldev->dev = &pdev->dev;
+	ldev->gpu_type = (enum lima_gpu_type)of_device_get_match_data(&pdev->dev);
 
 	platform_set_drvdata(pdev, ldev);
 
@@ -281,7 +282,7 @@ static int lima_pdev_remove(struct platform_device *pdev)
 }
 
 static const struct of_device_id dt_match[] = {
-	{ .compatible = "arm,mali-400" },
+	{ .compatible = "arm,mali-400", .data = (void *)GPU_MALI400 },
 	{}
 };
 MODULE_DEVICE_TABLE(of, dt_match);
